app/jinja_filters.py

```diff
@@ -43,6 +43,40 @@ def format_date(value):
     return value.strftime('%-d %B %Y')
 
 
+@blueprint.app_template_filter()
+def format_conditional_date(date1=None, date2=None):
+    """
+    This function format_conditional_date accepts two dates, a user submitted date and a metadata date
+
+    :param dates: a list of dates first one is user entered, second is metadata date
+    :return: the value of the date to be piped
+    """
+    dates = ([date1, date2])
+
+    if not len(dates):
+        raise Exception("Empty dates passed to format_conditional_dates filter")
+
+    if any(isinstance(item, int) for item in dates):
+        raise Exception("Invalid dates passed to format_conditional_dates filter")
+
+    if date1:
+        date = date1
+    else:
+        date = date2
+
+    if date is None:
+        raise Exception("No valid dates passed to format_conditional_dates filter")
+
+    d = str(date)
+    if len(d) < 8:
+        raise Exception("Date too short passed to format_conditional_dates filter")
```

Comment:
This check is only relevant for strings, but placing it here will also check Date objects. Moving it to the else below would be better.